### PR TITLE
Add neighbors from subzones to parent zone

### PR DESCRIPTION
**Pull request by FelixDQ**
From https://github.com/electricitymap/electricitymap-contrib/pull/4177

Needed to compute spatial aggregates.

### CODE REVIEW

File: electricitymap/contrib/config/__init__.py [hunk 1]
```diff
@@ -38,6 +45,26 @@
...
         ZONE_NEIGHBOURS[zone_name_1].add(zone_name_2)
+
+# Find neighbors to subzones and add them to parent zone.
+for zone in ZONES_CONFIG.keys():
+    subzones = ZONES_CONFIG[zone].get("subZoneNames", [])
+    if not subzones:
+        continue
+
+    for subzone in subzones:
+        for subzone_neighbor in ZONE_NEIGHBOURS[subzone]:
```
Comment (pierresegonne) [inline electricitymap/contrib/config/__init__.py:56]: isn't there a risk of iterating on `None`? 
Better to do `ZONE_NEIGHBOURS.get(subzone, [])`?

File: electricitymap/contrib/config/__init__.py [hunk 2]
```diff
@@ -38,6 +45,26 @@
...
+            if subzone_neighbor in subzones:
+                # ignore the neighbours that are within the zone
+                continue
+
+            if zone not in ZONE_NEIGHBOURS:
+                ZONE_NEIGHBOURS[zone] = set()
+
+            # If neighbor zone is a subzone, we add the exchange to the parent zone
```
Comment (pierresegonne) [inline electricitymap/contrib/config/__init__.py:64]: Image having `A-1`, `A-2`, `A-3`, `B-1`, `B-2`, `C` zones with exchanges between `A-1->A-2`, `A-2->B-1` and `A-2->C`. What would be the resulting neighbour relations?

Comment (FelixDQ) [reply electricitymap/contrib/config/__init__.py:64]: `A->B` and `A->C`

File: electricitymap/contrib/config/__init__.py (hunk 1 above)
Comment (FelixDQ) [reply electricitymap/contrib/config/__init__.py]: Yes, I think you are right. I changed it

**Review by pierresegonne — APPROVED** [review]
Nice

**Pull request merged**